src/internal_events/postgresql_metrics.rs

```diff
@@ -36,6 +36,13 @@ impl<'a> InternalEvent for PostgresqlMetricsCollectFailed<'a> {
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => self.error.to_string(),
+            "error_type" => "request_error",
```

Comment:
Can we add the `error_type` and `stage` to the associated error log as well?

Reply:
I missed it, thank you 😉 
To answer your question on `BytesReceived`, to compute the `byte_size`, we'd need to fetch all the values and it would be the same than building `EventsReceived`. I could call both but I don't really see the point and it wouldn't comply with the RFC saying it should be called before building the event.